src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -33,6 +33,7 @@
 import com.openshift.restclient.http.IHttpConstants;
 import com.openshift.restclient.utils.SSLUtils;
 
+import okhttp3.Authenticator;
```

Comment:
+1, I dont think that it's a good idea to expose the client library that we use. This would disallows us from moving off it without breaking API compat.
Thinking about an abstraction, I looked up a few of the most widely used http client libs. Shouldnt be too hard to provide an abstraction that's capable of using the following libs behind the scenes:
- okHttp: https://square.github.io/okhttp/3.x/okhttp/okhttp3/Authenticator.html
- apache httpclient: http://hc.apache.org/httpclient-3.x/apidocs/org/apache/commons/httpclient/auth/AuthScheme.html
- jdk HttpUrlConnection: https://docs.oracle.com/javase/7/docs/api/java/net/Authenticator.html
- etc.